### ISSUE TITLE
vivado-hls generate-project fails with unclear error message when required variables are missing

### ISSUE THREAD

**Issue by dinyar** [ipbus/ipbb#129]
When calling `ipbb vivado-hls generate-project` without 'device_package', 'device_name', and 'device_speed' the error message emitted is
```
ERROR ('TypeError' exception caught): 'can only join an iterable'

File "/afs/cern.ch/work/d/dinyar/ipbb-dev-2021b/src/ipbb/cmds/vivadohls.py", line 94, in genproject
   "Error caught while generating Vivado TCL commands:\n" + "\n".join(lExc),
```

Printing `lExc` manually then gives
```
Missing required variables: {'device_package', 'device_name', 'device_speed'}
```

**Comment by alessandrothea** [comment]
Hi,
Should be fixed with the last PR (#130 if I'm not mistaken) and in tag `dev/2021c`.

**Comment by alessandrothea** [comment]
@dinyar I'm closing this now. Please reopen the issue if the problem persists.
